cmd/bootstrap/cmd/finalize.go

```diff
@@ -634,3 +641,28 @@ func generateEmptyExecutionState(
 	log.Info().Msg("")
 	return
 }
+
+// validateEpochConfig validates configuration of the epoch commitment deadline.
+func validateEpochConfig() {
+	chainID := parseChainID(flagRootChain)
+	dkgFinalView := flagNumViewsInStakingAuction + flagNumViewsInDKGPhase*3 // 3 DKG phases
+	epochCommitDeadline := flagNumViewsInEpoch - flagEpochCommitSafetyThreshold
+
+	defaultSafetyThreshold, err := protocol.DefaultEpochCommitSafetyThreshold(chainID)
+	if err != nil {
+		log.Fatal().Err(err).Msg("unable to validate epoch config")
+	}
+
+	// sanity check 1: the safety threshold is >= the default for the chain
+	if flagEpochCommitSafetyThreshold < defaultSafetyThreshold {
+		log.Fatal().Msgf("potentially unsafe epoch config: epoch commit safety threshold smaller than expected (%d < %d)", flagEpochCommitSafetyThreshold, defaultSafetyThreshold)
+	}
+	// sanity check 2: the difference between DKG end and safety threshold is also >= the default safety threshold
```

Comment:
I think it's misplaced, should be moved to next if statement

Reply:
Yeah, I meant the two conditionals below taken together - but agree it's confusing. Updating